File: python/896.monotonic-array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:

        le, ge = True, True
        for n1, n2 in pairwise(nums):
            if n1 < n2:
                ge = False
            
            if n1 > n2:
                le = False

            if not ge and not le:
                break

        return le or ge

File: golang/896.monotonic-array.go
func isMonotonic(nums []int) bool {
    le, ge := true, true

    for i := 0; i < len(nums) - 1; i++ {
        if nums[i] < nums[i + 1] {
            ge = false
        }

        if nums[i] > nums[i + 1] {
            le = false
        }

        if !le && !ge {
            break
        }
    }

    return ge || le
}